modules/nf-commons/src/main/nextflow/Global.groovy

```diff
@@ -165,6 +165,17 @@ class Global {
 
     }
 
+    static List<String> getAzCredentials(Map env, Map config) {
+
+        String a
+
+        if( env && (a=env.AZURE_STORAGE_ACCOUNT_KEY) )  {
+            log.debug "Using AZ credentials defined by environment variables AZURE_STORAGE_ACCOUNT_KEY"
+            return [a]
+        }
+        return null
+    }
+
```

Comment:
This is an ugly legacy pattern that's used for the AWS credentials I'm trying to get rid in upcoming version. Let's not do the same for Azure.

Azure configuration can be retrieved via [AzConfig.config](https://github.com/nextflow-io/nextflow/blob/025ff9d0cc9d67a923896d58f89d8b2ac25defa1/plugins/nf-azure/src/main/nextflow/cloud/azure/config/AzConfig.groovy#L71-L73)

Reply:
@pablo-aledo there are couple of comments to address